Sources/SpeziAccount/AccountHeader.swift

```diff
@@ -0,0 +1,98 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SpeziViews
+import SwiftUI
+
+
+/// A summary view for ``SpeziAccountOverview`` that can be used as a Button to link to ``SpeziAccountOverview``.
+///
+/// Below is a short code example on how to use the `AccountHeader` view.
+///
+/// ```swift
+/// struct MyView: View {
+///     var body: some View {
+///         NavigationStack {
+///             Form {
+///                 Section {
+///                     NavigationLink {
+///                         AccountOverview()
+///                     } label: {
+///                         AccountHeader(details: details)
+///                     }
+///                 }
+///             }
+///         }
+///     }
+/// }
+/// ```
+public struct AccountHeader: View {
+    @EnvironmentObject private var account: Account
+    private var caption: LocalizedStringResource
+    
+    public var body: some View {
+        let accountDetails = account.details
+        
+        HStack {
+            UserProfileView(name: accountDetails?.name ?? PersonNameComponents(givenName: "Placeholder", familyName: "Placeholder"))
+                .frame(height: 60)
+                .redacted(reason: account.details == nil ? .placeholder : [])
+                .accessibilityHidden(true)
+            VStack(alignment: .leading) {
+                Text(accountDetails?.name?.formatted() ?? "Placeholder")
+                    .font(.title2)
+                    .fontWeight(.semibold)
+                    .redacted(reason: account.details == nil ? .placeholder : [])
+                Text(caption)
+                    .font(.caption)
+            }
+        }
+    }
+    
+    /// Display a new Account Header.
+    /// - Parameter caption: A descriptive text displayed under the account name giving the user a brief explanation of what to expect when they interact with the header.
+    public init(caption: LocalizedStringResource? = nil) {
+        self.caption = caption ?? LocalizedStringResource("ACCOUNT_HEADER_CAPTION", bundle: .atURL(from: .module))
+    }
+}
```

Comment:
A nice suggestion to clearly mark the caption as needed and deal with default values in SwiftUI views.
If you don't want that to clutter up your public documentation, you can use `@_documentation(visibility: internal)` to hide it from there:
```swift
public struct AccountHeader: View {
    /// Default values for the ``AccountHeader`` view.
    @_documentation(visibility: internal)
    public enum Defaults {
        /// Default caption.
        @_documentation(visibility: internal)
        public static let caption = LocalizedStringResource("ACCOUNT_HEADER_CAPTION", bundle: .atURL(from: .module))
    }
    
    // ...
}
```
```suggestion
    public init(caption: LocalizedStringResource = Defaults.caption) {
        self.caption = caption
    }
}
```
```

Reply:
Thanks for the neat suggestion @PSchmiedmayer!